src/array.jl

```diff
@@ -967,6 +967,12 @@ end
 Array(A::CategoricalArray{T}) where {T} = Array{T}(A)
 collect(A::CategoricalArray) = copy(A)
 
+# Defined for performance
+function collect(x::Base.SkipMissing{<: CatArrOrSub{T}}) where {T}
+    r = refs(x.x)
+    return CategoricalVector{T}(r[r .> 0], copy(pool(x.x)))
```

Comment:
this code is incorrect for 0-dimensional arrays or views:
```
julia> y = categorical(fill("a"))
0-dimensional CategoricalArrays.CategoricalArray{String,0,UInt32}:
"a"

julia> y[y.refs .> 0]
ERROR: ArgumentError: invalid index: true of type Bool
```
you have to special case this scenario. Other than that it looks good.

Reply:
Good catch. I think `filter` should work in all cases, right?

I also realized I forgot to call `nonmissingtype`.